File: 1-memento/java/example-word-editor-v1/Editor.java
public class Editor {
    private String content;
    private String fontName;
    private Integer fontSize;

    public String getContent(){
        return this.content;
    }

    public void print(){
        System.out.println("content: '" + this.content + "', fontName: '" + this.fontName + "', fontSize: '" + this.fontSize + "'" );
    }

    public void setContent(String content){
        this.content = content;
    }

    public String getFontName(){
        return this.fontName;
    }

    public void setFontName(String fontName){
        this.fontName = fontName;
    }

    public Integer getFontSize(){
        return this.fontSize;
    }

    public void setFontSize(Integer fontSize){
        this.fontSize = fontSize;
    }

    public EditorState createState(){
        return new EditorState(this.content, this.fontName, this.fontSize);
    }

    public void restoreState(EditorState state){
        this.content = state.getContent();
        this.fontName = state.getFontName();
        this.fontSize = state.getFontSize();
    }
}

File: 1-memento/js/example-word-editor-v1/Editor.js
const EditorState = require("./EditorState.js")

module.exports = class Editor {
    #content = null;
    #fontName = null;
    #fontSize = null;

    setState(content, fontName, fontSize) {
        this.#content = content;
        this.#fontName = fontName;
        this.#fontSize = fontSize;
    }
    printState() {
        console.log(`content: ${this.#content}, fontName: ${this.#fontName}, fontSize: ${this.#fontSize}`);
    }
    createEditorState() {
        return new EditorState(this.#content, this.#fontName, this.#fontSize);
    }
    restoreEditorState(editorState) {
        if (!(editorState instanceof EditorState)) throw new Error("editorState must be a EditorState");
        this.#content = editorState?.getContent();
        this.#fontName = editorState?.getFontName();
        this.#fontSize = editorState?.getFontSize();
    }
}